consensus/validator.go

```diff
@@ -316,3 +316,14 @@ func (consensus *Consensus) onCommitted(msg *msg_pb.Message) {
 	}
 	consensus.consensusTimeout[timeoutConsensus].Start()
 }
+
+func (consensus *Consensus) spinUpStateSync() {
+	select {
+	case consensus.BlockNumLowChan <- struct{}{}:
+		consensus.current.SetMode(Syncing)
+		for _, v := range consensus.consensusTimeout {
+			v.Stop()
+		}
+	default:
```

Comment:
what if the reader are not reading. Then this part will block forever right? potential goroutine leak. The original code has a timeout

Reply:
If `consesus.BlockNumLowChan` is occupied, the select will go to default, and the function will return immediately. Also, the function is not called with a new goroutine. So it shall not be not a problem as far as I see.

Reply:
ok, right.